ruby/bindings/InitRubyBindings.cpp

```diff
@@ -749,6 +753,21 @@ ENV.reject! { |k, _| ['GEM', 'BUNDLE'].any? { |x| k.start_with?(x) } }
     end
   end
 
+#  module Gem
+#    def self.default_dir
+#      @default_dir = ""
+#    end
+#
+#    def self.find_home
+#      ""
+#    end
+#
+#    def self.default_specifications_dir
+#      @default_specifications_dir = ""
+#    end
+#  end
+#
+  Gem.clear_paths
```

Comment:
This is going to force recomputing Gem @paths

Could have done `Gem.paths = {'GEM_HOME' => xxx, 'GEM_PATH' => xxx, 'GEM_SPEC_CACHE' => xx}`, but this is much better.

RubyEngine::setEmbeddedGems does:

* setupEmbeddedGemsClearEnvVars: set all three vars in ENV to ''
* initEmbeddedGems: will load rubygems
* setGemPathDir, setRubyEnvVarPath(various including GEM_HOME): all of these tweak ENV only if the path ins't empty, so it'll stay a '' blank string if not supplied
* locateEmbeddedGems calls Gem.clear_paths here. So paths is recomputed using the ENV variables via path_support, and will pick up the variables that were added (or not) to do so
*

Reply:
https://github.com/ruby/ruby/blob/e51014f9c05aa65cbf203442d37fef7c12390015/lib/rubygems.rb#L313-L323

```ruby
  ##
  # Reset the +dir+ and +path+ values.  The next time +dir+ or +path+
  # is requested, the values will be calculated from scratch.  This is
  # mainly used by the unit tests to provide test isolation.


  def self.clear_paths
    @paths         = nil
    @user_home     = nil
    Gem::Specification.reset
    Gem::Security.reset if defined?(Gem::Security)
  end
```

https://github.com/ruby/ruby/blob/e51014f9c05aa65cbf203442d37fef7c12390015/lib/rubygems.rb#L358-L363

```ruby
  # Retrieve the PathSupport object that RubyGems uses to
  # lookup files.


  def self.paths
    @paths ||= Gem::PathSupport.new(ENV)
  end
```